llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1489,7 +1489,7 @@ static void transformRecipestoEVLRecipes(VPlan &Plan, VPValue &EVL) {
                 Ops.push_back(&EVL);
                 return new VPWidenIntrinsicRecipe(Intrinsic::vp_select, Ops,
                                                   TypeInfo.inferScalarType(Sel),
-                                                  false, false, false);
+                                                  Sel->getDebugLoc());
```

Comment:
this also sets the debug info where it didn't set it before which is good to fix as well, thanks.

I think we need at least some tests for EVL vectorization with debug locations, to make sure they are preserved properly

Reply:
Yes, maybe need to create new tests similar to `llvm/test/Transforms/LoopVectorize/debugloc.ll` later, I think.